batched/dense/unit_test/Test_Batched_SerialTbsv.hpp

```diff
@@ -0,0 +1,340 @@
+      : _a(a), _b(b), _k(k) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const ParamTagType &, const int k) const {
+    auto aa = Kokkos::subview(_a, k, Kokkos::ALL(), Kokkos::ALL());
+    auto bb = Kokkos::subview(_b, k, Kokkos::ALL());
+
+    KokkosBatched::SerialTbsv<
+        typename ParamTagType::uplo, typename ParamTagType::trans,
+        typename ParamTagType::diag, AlgoTagType>::invoke(aa, bb, _k);
+  }
+
+  inline void run() {
+    using value_type = typename AViewType::non_const_value_type;
+    std::string name_region("KokkosBatched::Test::SerialTbsv");
+    const std::string name_value_type = Test::value_type_name<value_type>();
+    std::string name                  = name_region + name_value_type;
+    Kokkos::Profiling::pushRegion(name.c_str());
+    Kokkos::RangePolicy<execution_space, ParamTagType> policy(0, _b.extent(0));
+    Kokkos::parallel_for(name.c_str(), policy, *this);
+    Kokkos::Profiling::popRegion();
+  }
+};
+
+template <typename DeviceType, typename ScalarType, typename LayoutType,
+          typename ParamTagType, typename AlgoTagType>
+/// \brief Implementation details of batched tbsv test
+///
+/// \param N [in] Batch size of RHS (banded matrix can also be batched matrix)
+/// \param k [in] Number of superdiagonals or subdiagonals of matrix A
+/// \param BlkSize [in] Block size of matrix A
+void impl_test_batched_tbsv(const int N, const int k, const int BlkSize) {
+  using execution_space = typename DeviceType::execution_space;
+  using View2DType = Kokkos::View<ScalarType **, LayoutType, execution_space>;
+  using View3DType = Kokkos::View<ScalarType ***, LayoutType, execution_space>;
+
+  // Reference is created by trsv from triangular matrix
+  View3DType A("A", N, BlkSize, BlkSize), Ref("Ref", N, BlkSize, BlkSize);
+  View3DType Ab("Ab", N, k + 1, BlkSize);                 // Banded matrix
+  View2DType x0("x0", N, BlkSize), x1("x1", N, BlkSize);  // Solutions
+
+  Kokkos::Random_XorShift64_Pool<execution_space> random(13718);
+  Kokkos::fill_random(Ref, random, ScalarType(1.0));
+  Kokkos::fill_random(x0, random, ScalarType(1.0));
+
+  Kokkos::fence();
+
+  Kokkos::deep_copy(x1, x0);
+
+  // Create triangluar or banded matrix
+  create_banded_triangular_matrix<View3DType, View3DType,
+                                  typename ParamTagType::uplo>(Ref, A, k,
+                                                               false);
+  create_banded_triangular_matrix<View3DType, View3DType,
+                                  typename ParamTagType::uplo>(Ref, Ab, k,
+                                                               true);
+
+  // Reference trsv
+  Functor_BatchedSerialTrsv<DeviceType, View3DType, View2DType, ScalarType,
+                            ParamTagType, Algo::Trsv::Unblocked>(1.0, A, x0)
+      .run();
+
+  // tbsv
+  Functor_BatchedSerialTbsv<DeviceType, View3DType, View2DType, ParamTagType,
+                            AlgoTagType>(Ab, x1, k)
+      .run();
+
```

Comment:
Maybe fence after launching the kernel to be sure it has completed by the time you perform the check? parallel_for is non-blocking in general